### PR TITLE
feat(VBtnToggle): port to v3

### LINKED ISSUE #13321: [Feature Request] [v-btn-toggle] new disabled prop that propagate to child v-btn
Hello !

Here is a little thing that would be cool :)

### Problem to solve
If I want all the `v-btn` in a `v-btn-toggle` to be disabled, I have to set the same condition on all buttons. In order to 'DRY', I'd like to be able to set it only once on the `v-btn-toggle` component so that all the child `v-btn` get it.

A bit like the 'disabled' on a `v-form` component that propagate it to all child input components.

### Proposed solution
Add a `disabled` prop on the `v-btn-toggle` that would propagate to all child `v-btn` by default.

### LINKED ISSUE #13468: [Task] v-btn-toggle
# Overview

- Related Tasks: n/a
- Related Issues: n/a

A conversion should not start until *all* assigned issues with the milestone v2.2.x are resolved and in the `next` branch. Reference our [Coding Guidelines](https://github.com/vuetifyjs/vuetify/wiki/coding-guidelines) for information regarding the team's practices or reach out to us on [Discord](https://community.vuetifyjs.com).

### PR DESCRIPTION
**Pull request by johnleider**
## Description
fixes #13321 
fixes #13468

Added new styles to `v-btn-group` and use in `v-btn-toggle`. Updated `v-btn`  for group support and added the ability to use `useGroupItem` if nothing is injected.

## Motivation and Context
port to v3

## Markup:

<details>

```vue
<template>
  <v-app>
    <div class="ma-4 pa-4 bg-grey-lighten-3">
      <div class="d-flex">
        <v-checkbox v-model="one" label="Elevation" hide-details />
        <v-checkbox v-model="two" label="Border" hide-details />
        <v-checkbox v-model="three" label="Tile" hide-details />
      </div>

      <div class="d-flex">
        <v-checkbox v-model="four" label="Color" hide-details />
        <v-checkbox v-model="divided" label="Divided" hide-details />
      </div>

      <br>

      {{ model }}

      <br>

      <h3>Default</h3>
      <v-btn-toggle
        v-model="model"
        :border="two"
        :color="four ? 'primary' : undefined"
        :elevation="one ? 6 : undefined"
        :tile="three"
        :divided="divided"
        variant="text"
      >
        <v-btn icon="mdi-vuetify" />
        <v-btn icon="mdi-format-align-center" />
        <v-btn icon="mdi-format-align-right" />
        <v-btn icon="mdi-format-align-justify" />
        <v-btn prepend-icon="mdi-vuetify">
          Inline text
        </v-btn>
      </v-btn-toggle>

      <br>
      <br>

      <h3>Comfortable</h3>
      <v-btn-group
        :border="two"
        :color="four ? 'primary' : undefined"
        :elevation="one ? 6 : undefined"
        :tile="three"
        :divided="divided"
        density="comfortable"
      >
        <v-btn icon="mdi-vuetify" />
        <v-btn icon="mdi-format-align-center" />
        <v-btn icon="mdi-format-align-right" />
        <v-btn icon="mdi-format-align-justify" />
        <v-btn prepend-icon="mdi-vuetify">
          Inline text
        </v-btn>
      </v-btn-group>

      <br>
      <br>

      <h3>Compact</h3>
      <v-btn-group
        :border="two"
        :color="four ? 'primary' : undefined"
        :elevation="one ? 6 : undefined"
        :tile="three"
        :divided="divided"
        density="compact"
      >
        <v-btn icon="mdi-vuetify" />
        <v-btn icon="mdi-format-align-center" />
        <v-btn icon="mdi-format-align-right" />
        <v-btn icon="mdi-format-align-justify" />
        <v-btn prepend-icon="mdi-vuetify">
          Inline text
        </v-btn>
      </v-btn-group>
    </div>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
      divided: false,
      model: null,
      one: false,
      two: false,
      three: false,
      four: false,
      five: false,
    }),
  }
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
Requesting **Initial** comments from @vuetifyjs/core-team

File: packages/vuetify/src/components/VBtn/VBtn.tsx [hunk 1]
```diff
@@ -81,10 +84,11 @@ export const VBtn = defineComponent({
           type={ Tag === 'a' ? undefined : 'button' }
           class={[
             'v-btn',
+            group?.selectedClass?.value,
```
Comment (KaelWD) [inline packages/vuetify/src/components/VBtn/VBtn.tsx:87]:
```suggestion
            group?.selectedClass.value,
```

File: packages/vuetify/src/composables/group.ts [hunk 2]
```diff
@@ -73,7 +73,8 @@ export const makeGroupItemProps = propsFactory({
 export function useGroupItem (
   props: { value?: unknown, disabled?: boolean, selectedClass?: string },
   injectKey: InjectionKey<GroupProvide>,
-): GroupItemProvide {
+  required = true
+): GroupItemProvide | null {
```
Comment (KaelWD) [inline packages/vuetify/src/composables/group.ts:77]:
```ts
export function useGroupItem<R extends boolean> (
  props: { value?: unknown, disabled?: boolean, selectedClass?: string },
  injectKey: InjectionKey<GroupProvide>,
  required: R = true
): R extends true ? GroupItemProvide : GroupItemProvide | null {
```

**Comment by KaelWD** [comment]
I reverted the expansion panel changes and deleted VBtnToggle.old.ts

**Pull request merged**